torch/_refs/__init__.py

```diff
@@ -2055,6 +2057,64 @@ def tensor_split(
         return tuple(splits)
 
 
+def hsplit(
+    a: TensorLikeType, indices_or_sections: DimsType
+) -> Tuple[TensorLikeType, ...]:
+    msg = (
+        "torch.hsplit requires a tensor with at least 1 dimension, but got a tensor with "
+        + str(a.ndim)
+        + " dimensions!"
+    )
+    check(a.ndim >= 1, lambda: msg)
+    dim = 0 if a.ndim == 1 else 1
+    if isinstance(indices_or_sections, int):
+        split_size = indices_or_sections
+        msg = (
```

Comment:
Doing final checks -- what I don't like about this pattern is that we're constructing the message even when we don't error, cc @ezyang -- I think we have to check for the condition to construct the message only in the error state, then if we want to use `check` just `check(False, ...`

Reply:
(the other thing to do is we could define a function that returns the string, which would also delay string construction and which `check` does accept. I'm not sure how costly defining a function is, however -- I'd prefer we hide as much as possible behind the error checks

Reply:
This is why check takes a lambda; `check(b, lambda: string_construction)`. Taking a string directly is wrong and won't actually work. I checked and dynamo is able to trace past the lambda construction.